crates/core/tedge_write/tests/main.rs

```diff
@@ -31,49 +33,59 @@ fn creates_dest_file_if_doesnt_exist() {
 }
 
 #[test]
-fn sets_file_permissions_if_file_doesnt_exist() {
+fn changes_permissions_if_destination_doesnt_exist() {
     // Arrange
     let (temp_dir, source_path) = setup_source_file();
-    fs::set_permissions(&source_path, Permissions::from_mode(0o644)).unwrap();
+    chown(&source_path, Some(2137), Some(2137)).unwrap();
```

Comment:
Such a test can only be successful if run by a privileged user allowed to `chown` a file.

Reply:
Rust integration tests were a bad environment for such a test, they were moved to RobotFramework in 43f5705.